File: fizzbuzz_initial/fizzbuzz.js
var fizzbuzz = function(number) {
  // returns 'fizz' if number is divisible by 5,
  // 'buzz' if number is divisible by 7,
  // and 'fizzbuzz' if number is divisible by 35
  var output = '';
  output += fizz(number);
  output += buzz(number);
  return output;
};

var fizz = function(number) {
  if (number % 5 === 0) {
    return 'fizz';
  }
  return '';
}

var buzz = function(number) {
  if (number % 7 === 0) {
    return 'buzz';
  }
  return '';
}

console.log(fizzbuzz(3) === '')
console.log(fizzbuzz(5) === 'fizz')
console.log(fizzbuzz(7) === 'buzz')
console.log(fizzbuzz(28) === 'buzz')
console.log(fizzbuzz(25) === 'fizz')
console.log(fizzbuzz(49) === 'buzz')
console.log(fizzbuzz(35) === 'fizzbuzz')

File: fizzbuzz_initial/fizzbuzz.rb
class Integer
  def fizzbuzz
    output = ''
    output << self.fizz
    output << self.buzz
  end

  def fizz
    self % 5 == 0 ? 'fizz' : ''
  end

  def buzz
    self % 7 == 0 ? 'buzz' : ''
  end
end

puts 3.fizzbuzz == ''
puts 5.fizzbuzz == 'fizz'
puts 7.fizzbuzz == 'buzz'
puts 35.fizzbuzz == 'fizzbuzz'
puts 0.fizzbuzz == 'fizzbuzz'
puts 21.fizzbuzz == 'buzz'
puts 49.fizzbuzz == 'buzz'
puts 77.fizzbuzz == 'buzz'
puts 105.fizzbuzz == 'fizzbuzz'
puts 25.fizzbuzz == 'fizz'
puts 100.fizzbuzz == 'fizz'
puts 50.fizzbuzz == 'fizz'
